velox/type/DecimalUtil.h

```diff
@@ -301,6 +302,79 @@ class DecimalUtil {
     return remainder * resultSign;
   }
 
+  /// Returns the max required size to convert the decimal of this precision and
+  /// scale to varchar. A varchar's size is estimated with unscaled value
+  /// digits, dot, leading zero, and possible minus sign.
+  static int32_t maxStringViewSize(int precision, int scale) {
+    int32_t rowSize = precision + 1; // Number and symbol.
+    if (scale > 0) {
+      ++rowSize; // A dot.
+    }
+    if (precision == scale) {
+      ++rowSize; // Leading zero.
+    }
+    return rowSize;
+  }
+
+  /// @brief Convert the unscaled value of a decimal to varchar and write to raw
+  /// string buffer from start position.
+  /// @tparam T The type of input value.
+  /// @param unscaledValue The input unscaled value.
+  /// @param scale The scale of decimal.
+  /// @param maxVarcharSize The estimated max size of a varchar.
+  /// @param startPosition The start position to write from.
+  /// @return A string view.
+  template <typename T>
+  static StringView convertToStringView(
+      T unscaledValue,
+      int32_t scale,
+      int32_t maxVarcharSize,
+      char* const startPosition) {
+    char* writePosition = startPosition;
+    if (unscaledValue == 0) {
+      *writePosition++ = '0';
+      if (scale > 0) {
+        *writePosition++ = '.';
+        // Append leading zeros.
```

Comment:
Is this comment accurate? It seems like it is adding trailing zeros, not leading.

Reply:
It is appending zeros before the fractional digit like '1.00003', so I used 'leading zeros' here.